be/src/vec/core/types.h

```diff
@@ -325,6 +341,44 @@ struct Decimal {
         return *this;
     }
 
+    std::string to_string(UInt32 scale) const {
+        static constexpr auto precision =
+                std::is_same_v<T, Int32> ? 9 : (std::is_same_v<T, Int64> ? 18 : 38);
```

Comment:
Use `BeConsts::MAX_DECIMAL32_PRECISION / MAX_DECIMAL64_PRECISION / MAX_DECIMAL128_PRECISION` to replace `9 / 18 / 38`